chainer/links/loss/hierarchical_softmax.py

```diff
@@ -299,19 +310,48 @@ def __init__(self, in_size, tree):
         # This function object is copied on every forward computation.
         super(BinaryHierarchicalSoftmax, self).__init__()
         self._func = BinaryHierarchicalSoftmaxFunction(tree)
+        self.tree = tree
+        self.PAD_paths, self.PAD_codes = self.init_padding_params()
```

Comment:
I feel it is enough to assign these variables in `init_padding_params` instead of returning values.